ns1/record_helper.go

```diff
@@ -14,14 +14,17 @@ type RecordMutexKV struct {
 	tracker map[string][]interface{}
 }
 
-func (m *RecordMutexKV) Lock(client *ns1.Client, i interface{}, record string, zone string) error {
+// Lock creates a TXT record in the given zone indicating that the given record is being altered, so that parallel
+// executions of this provider are synchronized. This is safe for use by multiple goroutines. A non-nil error is
+// returned if the TXT record cannot be created.
```

Comment:
If the record is already locked, does it return an error, or does it block until the record can be locked?